stripe/src/main/java/com/stripe/android/model/SourceParams.java

```diff
@@ -348,6 +348,27 @@ public String getType() {
         return mType;
     }
 
+    /**
+     * Create a string-keyed map representing this object that is
+     * ready to be sent over the network.
+     *
+     * @return a String-keyed map
+     */
+    @NonNull
+    public Map<String, Object> toParamMap() {
+        Map<String, Object> networkReadyMap = new HashMap<>();
+        networkReadyMap.put(API_PARAM_TYPE, getType());
+        networkReadyMap.put(getType(), getApiParameterMap());
+        if (!(NO_AMOUNT_SOURCES.contains(getType()) && mAmount == 0L)) {
```

Comment:
we've handled the optionality of amount in the iOS SDK by boxing the amount in an object – maybe a java equivalent would make sense here? i'm a little concerned about the "no amount sources" logic – it's another thing to remember to update when adding a new source.